esphome/components/wifi/__init__.py

```diff
@@ -253,6 +254,7 @@ def _validate(config):
             cv.Optional(CONF_MANUAL_IP): STA_MANUAL_IP_SCHEMA,
             cv.Optional(CONF_EAP): EAP_AUTH_SCHEMA,
             cv.Optional(CONF_AP): WIFI_NETWORK_AP,
+            cv.Optional(CONF_ENABLE_IPV6): cv.boolean,
```

Comment:
This seems to only be implemented for ESP-IDF, so it should only be configurable when using IDF:
```suggestion
            cv.SplitDefault(CONF_ENABLE_IPV6, esp32_idf=False): cv.All(cv.only_with_esp_idf, cv.boolean),
```